### PR TITLE
fix: only reload model if has internet access

### PR DESCRIPTION
**Pull request by adong**
## Context

Every once in a while, users are offline from the internet or disconnected from VPN. Our users should be notified if they are offline, rather than being redirected to 404 page.

## Objective

This PR adds the warning only if it was an internet issue, and show screenshot like following:

image

## References

https://github.com/screwdriver-cd/screwdriver/issues/2325

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/pipeline/events/route.js [hunk 1]
```diff
@@ -26,8 +30,15 @@ export default Route.extend({
         count: ENV.APP.NUM_EVENTS_LISTED
       }),
       triggers: this.triggerService.getDownstreamTriggers(this.get('pipeline.id'))
-    }).catch(() => {
-      this.transitionTo('/404');
+    }).catch(err => {
+      if (err === '0 Request Failed') {
```
Comment (jithine) [inline app/pipeline/events/route.js:34]: is this a standard message across all browsers ? Any references ?

Comment (adong) [reply app/pipeline/events/route.js:34]: No this error message is generated by us. 
https://github.com/screwdriver-cd/ui/blob/b7d49559d63bd25060f4b0e9fa65e77c215421ee/app/pipeline-triggers/service.js#L29-L35

Screen Shot 2021-02-02 at 10 41 41 AM

File: app/pipeline/events/route.js [hunk 2]
```diff
@@ -3,6 +3,8 @@ import Route from '@ember/routing/route';
 import ENV from 'screwdriver-ui/config/environment';
 import RSVP from 'rsvp';
 
+const ERROR_MESSAGE = 'Session timed-out, please login back in to complete the action';
```
Comment (tkyi) [inline app/pipeline/events/route.js:6]:
```suggestion
const ERROR_MESSAGE = 'Session timed-out, please log back in to complete the action';
```

File: app/pipeline/jobs/index/route.js [hunk 3]
```diff
@@ -4,6 +4,8 @@ import ENV from 'screwdriver-ui/config/environment';
 import RSVP from 'rsvp';
 import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
 
+const ERROR_MESSAGE = 'Session timed-out, please login back in to complete the action';
```
Comment (tkyi) [inline app/pipeline/jobs/index/route.js:7]:
```suggestion
const ERROR_MESSAGE = 'Session timed-out, please log back in to complete the action';
```

Comment (tkyi) [reply app/pipeline/jobs/index/route.js:7]: or "please log in again"

**Review by jithine — APPROVED** [review]

**Comment by jithine** [comment]
Also @adong please test how it behaves when session expires, I believe 404 happens also when session expires.

**Comment by adong** [comment]
Add check to handle invalid token case.
image

**Review by jithine — COMMENTED** [review]
We seem to have a bunch of 404 routes https://github.com/screwdriver-cd/ui/search?q=404
are all  cases handled ?

**Comment by adong** [comment]
> We seem to have a bunch of 404 routes [screwdriver-cd/ui/search?q=404](https://github.com/screwdriver-cd/ui/search?q=404)
> are all cases handled ?

Some of them are valid 404 cases. Let me do another scan to make sure.

**Comment by adong** [comment]
These files look like have valid 404 cases
```
app/templates/detail/route.js
84:       if (error.status === 404) {

app/commands/detail/route.js
61:         this.transitionTo('/404');

app/pipeline/route.js
20:       if (error && Array.isArray(error.errors) && error.errors[0].status === 404) {

```

**Pull request merged**